pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -30,7 +30,7 @@ import (
 var log = logf.Log.WithName("controller_nstemplateset")
 
 // Add creates a new NSTemplateSetReconciler and starts it (ie, watches resources and reconciles the cluster state)
-func Add(mgr manager.Manager, _ *configuration.Config) error {
+func Add(mgr manager.Manager, _ *configuration.Config, _ client.Client) error {
```

Comment:
I understand the benefit of having the same `Add` func for all controllers, but it's a bit confusing that most of them do *not* use the given `client.Client`. Anyways, it's just my thoughts, not a request to change that.

Reply:
In the next PR, I'm going to use the client in the MemberStatus controller - that was also the reason why I was doing all the changes (you can read it in the description of the PR). So, there will be two controllers using this client.
BTW, you could raise the same concern regarding the `configuration.Config` param - this one is used only in two controllers as well. 
We are going to have four types of the constructors:
1. taking only Manager
2. taking Manager & Config
3. taking Manager & Client
4. taking Manager & Config & Client

So do you want to address all of them separately? Yeah, we could get rid of the `addToManagerFuncs` and call them directly in `AddControllersToManager`, but I'm probably too old for such changes :smile:

Reply:
yes, the `config` param raises the same concern. It's just that the `client` feels like something more "low-level" vs the `config` which is a type of ours. But let's keep it that way for now.